pkg/kafka/topicprovisioner.go

```diff
@@ -65,9 +73,15 @@ func NewTopicProvisioner(config TopicProvisionerConfig) (TopicProvisioner, error
 		return nil, errors.New("meter is required")
 	}
 
+	protectedTopics := make(map[string]struct{}, len(config.ProtectedTopics))
```

Comment:
nit: why isn't this bool like? It's a set, right?
`protectedTopicsByTopicName := map[string]bool`

Reply:
Yeah, it is a set, but empty `struct{}` does not require memory allocation while `bool` does. Thats why I prefer `map[string]struct{}` over `map[string]bool`.